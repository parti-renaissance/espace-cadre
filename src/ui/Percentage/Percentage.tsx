import PropTypes from 'prop-types'

interface PercentageProps {
  children: string | number
}
const Percentage = ({ children }: PercentageProps) => <>{`${(Number(children) * 100).toFixed(2)}%`}</>

Percentage.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
}

export default Percentage
