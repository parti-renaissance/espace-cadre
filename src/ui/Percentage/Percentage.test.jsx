import Percentage from 'ui/Percentage/Percentage'
import { render } from '@testing-library/react'

describe('Percentage', () => {
  it('displays percentage from text', () => {
    const { container } = render(<Percentage>0.123456789</Percentage>)

    expect(container).toMatchSnapshot()
  })

  it('displays percentage from number', () => {
    const { container } = render(<Percentage>{0.123456789}</Percentage>)

    expect(container).toMatchSnapshot()
  })
})
