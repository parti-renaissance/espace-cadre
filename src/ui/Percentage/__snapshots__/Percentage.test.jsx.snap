// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Percentage > displays percentage from number 1`] = `
<div>
  12.35%
</div>
`;

exports[`Percentage > displays percentage from text 1`] = `
<div>
  12.35%
</div>
`;
